monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/GroupByOperator.scala

```diff
@@ -121,10 +121,27 @@ private[reactive] final class GroupByOperator[A, K](
           }
       }
 
+      private[this] def errorAll(ex: Throwable): Seq[Throwable] = {
+        val cache = cacheRef.get()
+
+        if (!cacheRef.compareAndSet(cache, Map.empty))
+          errorAll(ex)
+        else
+          cache.values.foldLeft(Vector.empty[Throwable]) { (acc, o) =>
```

Comment:
I feel like this error accumulation (here and in `completeAll` ) shouldn't be necessary according to [Observer contract](https://monix.io/docs/3x/reactive/observers.html#contract)

Reply:
Agreed, fixed in https://github.com/monix/monix/pull/1190/commits/7140d7db336f56bd8302895d999fc5beab4b896a. (`errorAll` was almost a direct copy-paste of `completeAll`.)